R/utils.R

```diff
@@ -1057,3 +1057,50 @@ countAndPrint <- function(x) {
   if (inherits(object, "OnDiskMSnExp")) msLevel(object)[1]
   else msLevel(object[[1]])
 }
+
+#' @title Define the type of mzR backend to use based on the file name
+#'
+#' @description Simple helper to define the mzR backend that should/can be used
+#'     to read the file.
+#'
+#' @param x \code{character(1)} representing the file name.
+#'
+#' @return A \code{character(1)} with the name of the backend (either
+#'     \code{"netCDF"}, \code{"Ramp"} or \code{"pwiz"}.
+#' 
+#' @author Johannes Rainer, Sebastian Gibb
+#'
+#' @noRd
+.mzRBackend <- function(x = character()) {
+    if (length(x) != 1)
+        stop("parameter 'x' has to be of length 1")
+    ## Use if/else conditions based on a suggestion from sgibb to avoid loops.
+    if (grepl("\\.mzml($|\\.)|\\.mzxml($|\\.)", x, ignore.case = TRUE)) {
+        return("pwiz")
+    } else if (grepl("\\.mzdata($|\\.)", x, ignore.case = TRUE)) {
+        return("Ramp")
+    } else if (grepl("\\.cdf($|\\.)|\\.nc($|\\.)", x, ignore.case = TRUE)) {
+        return("netCDF")
+    } else {
+        stop("Could not determine file type for ", x)
+    }
+}
+
+#' @title Open an MS file using the mzR package
+#'
+#' @description Opens an MS file using the mzR package determining the corrent
+#'     backend based on the file ending of the specified file.
+#'
+#' @param x \code{character(1)}: the file name.
+#'
+#' @return A file handle to the opened MS file.
+#'
+#' @author Johannes Rainer
+#' 
+#' @noRd
+.openMSfile <- function(x) {
+    if(missing(x) || length(x) != 1)
```

Comment:
Oups, sorry, I thought it was done :-/. I think you can just commit that change.